File: hacker-rank/thirty-days-of-code/day-25/solution.js


function processData(input) {
  let array = input.split('\n')
  let T = array[0]
  for (let i = 1; i <= T; i++) {
    let n = array[i]
    if (isPrime(n)) {
      console.log(n + " Prime");
    }
    else {
      console.log(n + " Not Prime");
    }
  }
} 

function isPrime(n) {
  if (n == 0 || n == 1) return false;
  let limit = Math.floor(Math.sqrt(n));
  for (let i = 2; i <= limit; i++) {
    if (isPrime(i)) {
      if (n % i == 0) {
        return false;
      }
    }
  }
  return true;
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
  _input += input;
});

process.stdin.on("end", function () {
 processData(_input);
});

File: hacker-rank/thirty-days-of-code/day-25/Solution.java
import java.io.*;
import java.util.*;

public class Solution {

  private static boolean isPrime(int n) {
    if (n == 0 || n == 1) return false;
    int limit = (int) Math.floor(Math.sqrt((double) n));
    for (int i = 2; i <= limit; i++) {
      if (isPrime(i)) {
        if (n % i == 0) {
          return false;
        }
      }
    }
    return true;
  }

  public static void main(String[] args) {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
    Scanner sc = new Scanner(System.in);
    int T = sc.nextInt();

    for (int i = 0; i < T; i++) {
      int n = sc.nextInt();
      if (isPrime(n)) {
        System.out.println(n + " Prime");
      }
      else {
        System.out.println(n + " Not Prime");
      }
    }
    sc.close();
  }
}